samples: matter: Added snippet for schedules to Matter Lock sample (#16710)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

  * The :kconfig:option:`CONFIG_NCS_SAMPLE_MATTER_ZAP_FILES_PATH` Kconfig option, which specifies ZAP files location for the sample.
    By default, the option points to the :file:`src/default_zap` directory and can be changed to any path relative to sample's location that contains the ZAP file and :file:`zap-generated` directory.

* :ref:`matter_lock_sample` sample:

>   * Added :ref:`Matter Lock schedule snippet <matter_lock_snippets>`, and updated :file:`README.rst` file with instructions on how to enable the :ref:`scheduled timed access <matter_lock_enabling_scheduled_timed_access>` feature.

Reviewer: ```suggestion
  * Added :ref:`Matter Lock schedule snippet <matter_lock_snippets>`, and updated the documentation to use the snippet.
```
